File: fixtures/scale/ruby/large/lib/check_ferris.rb
require_relative 'ferris_encrypt'
require_relative 'ferris_decrypt'
require_relative 'corpus_rotate'

def check_ferris
  corpus_rotate.each do |text|
    sealed = ferris_encrypt(text, 5)
    opened = ferris_decrypt(sealed, 5)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_ferris.py
from cipherlab.ciphers.ferris import ferris_encrypt, ferris_decrypt
from cipherlab.specs.ferris_spec import ferris_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_ferris():
    spec = ferris_spec()
    for text in corpus_rotate():
        sealed = ferris_encrypt(text, spec["key"])
        opened = ferris_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"
